controllers/argocd/argocd_controller.go

```diff
@@ -105,6 +108,19 @@ func (r *ReconcileArgoCD) Reconcile(ctx context.Context, request ctrl.Request) (
 		return reconcile.Result{}, err
 	}
 
+	// Fetch labelSelector from r.LabelSelector (command-line option)
+	labelSelector, err := labels.Parse(r.LabelSelector)
+	if err != nil {
+		reqLogger.Info("error parsing the labelSelector '%s'.", labelSelector)
+		return reconcile.Result{}, err
+	}
+
+	// Match the value of labelSelector from ReconcileArgoCD to labels from the argocd instance
+	if !labelSelector.Matches(labels.Set(argocd.Labels)) {
+		reqLogger.Info("the ArgoCD instance does not match the label selector %s and skipping for reconcillation", r.LabelSelector)
+		return reconcile.Result{}, nil
+	}
+
 	newPhase := argocd.Status.Phase
```

Comment:
Can you add a log, saying the argocd instance is selected by the label selector for reconcillation and for default value, add a log message saying that all argocd instances are selected for reconcillation. This might help in debugging issues.